leapp-report-analyzer.py

```diff
@@ -0,0 +1,199 @@
+import argparse
+import glob
+import json
+import os
+import sys
+from collections import OrderedDict, defaultdict
+
+# Check if Python version is 3.6 or higher (cgi.escape is deprecated/removed for newer versions)
+if sys.version_info.major >= 3 and sys.version_info.minor >= 6:
+    from html import escape
+else:
+    from cgi import escape
+
+SEVERITY_ORDER = ["inhibitor", "high", "medium", "low", "info"]
+HTML_TABLE_OPTIONS = "border='1' cellpadding='5' cellspacing='0'"
+
+
+def _create_grouped_data_structure():
+    # JSON structure: {severity: [key: {title: str, {hostnames: [], remediations: {}}]}
+    return defaultdict(lambda: defaultdict(lambda: {"title": "", "hostnames": [], "remediations": []}))
+
+
+def escape_html(text):
+    return escape(text, quote=True)
+
+
+def main():
+    parser = get_parser()
+    args = parser.parse_args()
+
+    if args.input_directory:
+        input_files = glob.glob(os.path.join(args.input_directory, "*.json"))
+    else:
+        input_files = args.input_files
+
+    grouped_data_list = [parse_leapp_report(input_file) for input_file in input_files]
+    merged_grouped_data = merge_grouped_data(grouped_data_list)
+    ordered_merged_data = order_grouped_data(merged_grouped_data)
+    write_grouped_data_to_json(ordered_merged_data, args.output_json)
+    html_tables = generate_html_tables(ordered_merged_data)
+    write_html_tables_to_file(html_tables, args.output_html)
```

Comment:
Can we have some basic logging? At least some basic info what is happening during the execution.

Reply:
Added